video_creation/screenshot_downloader.py

```diff
@@ -105,8 +106,18 @@ def get_screenshots_of_reddit_posts(reddit_object: dict, screenshot_num: int):
             print_substep("Skipping translation...")
 
         postcontentpath = f"assets/temp/{reddit_id}/png/title.png"
-        page.locator('[data-test-id="post-content"]').screenshot(path=postcontentpath)
-
+        try :
+            page.locator('[data-test-id="post-content"]').screenshot(path=postcontentpath)
+        
+        except TimeoutError as e:
+            print_step("unable to locate post It is possibly Due to a NSFW post or unstable internet")
```

Comment:
Maybe create an if statement for this to check if NSFW is enabled?

Also, typo:

```suggestion
            print_step("Unable to get the post. Check your internet connection.")
```